libevmasm/AssemblyItem.cpp

```diff
@@ -195,14 +201,18 @@ size_t AssemblyItem::bytesRequired(size_t _addressLength, langutil::EVMVersion _
 
 size_t AssemblyItem::arguments() const
 {
-	if (hasInstruction())
+	if (hasInstruction() && type() != CallF && type() != JumpF && type() != RetF)
 		// The latest EVMVersion is used here, since the InstructionInfo is assumed to be
 		// the same across all EVM versions except for the instruction name.
 		return static_cast<size_t>(instructionInfo(instruction(), EVMVersion()).args);
 	else if (type() == VerbatimBytecode)
 		return std::get<0>(*m_verbatimBytecode);
 	else if (type() == AssignImmutable)
 		return 2;
+	else if (type() == CallF || type() == JumpF)
+		return functionSignature().argsNum;
+	else if (type() == RetF)
+		return static_cast<size_t>(data());
```

Comment:
Isn't this the number of returns? The number of arguments for `RETF` is zero.

```suggestion
		return 0;
```

Reply:
I still think this should be 0. See https://github.com/ethereum/solidity/pull/15550#discussion_r1857373561.

Reply:
Having the number of returns in `data()` on `RetF` is now redundant. We only use it in one random assert now. We should remove it. And the `_rets` argument from `AssemblyItem::functionReturn()` as well.